lib/stateView/document/mongoDbRepository/SVDocumentMongoDbRepository.js

```diff
@@ -27,10 +27,21 @@ class SVDocumentMongoDbRepository {
    * Find SVDocument by id
    *
    * @param {string} id
+   * @param {MongoDBTransaction} [stateViewTransaction]
    * @returns {Promise<SVDocument>}
    */
-  async find(id) {
-    const result = await this.mongoClient.findOne({ _id: id });
+  async find(id, stateViewTransaction) {
+    const findQuery = { _id: id };
+
+    let result;
+    if (stateViewTransaction) {
+      const session = stateViewTransaction.getSession();
+      const txFunc = async () => this.mongoClient.findOne(findQuery, { session });
```

Comment:
You need to use mongoClient and session from arguments. So you need to select `db` and `collection` as well.